pkg/getter/httpgetter.go

```diff
@@ -106,10 +107,13 @@ func NewHTTPGetter(options ...Option) (Getter, error) {
 }
 
 func (g *HTTPGetter) httpClient() (*http.Client, error) {
-	transport := &http.Transport{
-		DisableCompression: true,
-		Proxy:              http.ProxyFromEnvironment,
+	if g.transport == nil {
+		g.transport = &http.Transport{
```

Comment:
It may be worth creating the transport on an [once.Do() block](https://pkg.go.dev/sync#Once.Do) to avoid race conditions.